starter-code/public/javascript/APIHandler.js

```diff
@@ -1,25 +1,76 @@
 class APIHandler {
-  constructor (baseUrl) {
+  constructor(baseUrl) {
     this.BASE_URL = baseUrl;
   }
 
-  getFullList () {
+  // -- DISPLAY ALL -- //
+  getFullList() {
+    axios.get(`${this.BASE_URL}/characters`)
+      .then(chars => {
+        $('.characters-container-main').empty();
+        chars.data.forEach(item => {
+          displayOneCharacter(item); // (function in index.js)
+        })
+      })
+      .catch(err => {
+        console.log(err);
+      })
+  };
 
-  }
-
-  getOneRegister () {
+  // -- DISPLAY ONE -- //
+  getOneRegister(charId) {
+    axios.get(`${this.BASE_URL}/characters/${charId}`)
+      .then(char => {
+        $('.characters-container-main').empty();
+        displayOneCharacter(char.data); // (function in index.js)
+      })
+      .catch(err => {
+        console.log(err);
+      })
+  };
 
+  // -- CREATE ONE -- //
+  createOneRegister(characterObject) {
+    $('#send-data-create').removeClass('green-button red-button'); // reset status
+    axios.post(`${this.BASE_URL}/characters`, characterObject)
+      .then(result => { 
+        $('#send-data-create').addClass('green-button');
+      })
+      .catch(err => {
+        console.log(err);
+        $('#send-data-create').addClass('red-button');
+      })
   }
 
-  createOneRegister () {
-
+  // -- UPDATE ONE -- //
+  updateOneRegister(characterObject) {
+    $('#send-data-edit').removeClass('green-button red-button'); // reset status
+    axios.patch(`${this.BASE_URL}/characters/${characterObject.id}`, {
```

Comment:
A cleaner solution would be to create the object in the index.js (line 27 http://prntscr.com/n934ow) and pass in a ready object as an argument to this axios.patch(). 

This would result in a cleaner API handler function. Without the need to create the object > http://prntscr.com/n935eo